src/rest_vol_dataset.c

```diff
@@ -5217,6 +5369,370 @@ RV_get_cmpd_subset_nmembers(hid_t src_type_id, hid_t dst_type_id, size_t *num_cm
     if (dst_member_type > 0)
         if (H5Tclose(dst_member_type) < 0)
             FUNC_DONE_ERROR(H5E_DATATYPE, H5E_CANTCLOSEOBJ, FAIL, "can't close destination datatype member");
+}
+
+/*-------------------------------------------------------------------------
+ * Function:    RV_vlen_data_to_json
+ *
+ * Purpose:     Converts a buffer of hvl_t instances to a JSON value
+ *              which may be sent to the server as part of a write.
+ *
+ *              Allocates memory under the *out pointer on success which must be
+ *              freed by the calling function.
+ *
+ * Return:      Non-negative on success/Negative on failure
+ *
+ * Programmer:  Matthew Larson
+ *              January, 2024
+ */
+static herr_t
+RV_vlen_data_to_json(const void *_in, size_t nelems, hid_t dtype_id, void **out, size_t *out_size)
```

Comment:
Is `_in` meant to be anything other than a `const hvl_t *`? If not, just change this from `const void *` to `const hvl_t *` to get a little more type safety and remove the unnecessary casting below. Avoid passing `void *` anywhere that it isn't necessary.